lib/adapters/file/write.js

```diff
@@ -29,25 +30,95 @@ class WriteFile extends AdapterWrite {
 
         this.filename = options.file;
         this.format = options.format ? options.format : 'json'; // default to json
+
+        this.append = options.append;
+        this.serializerOptions = {};
+        if (options.append && !_.contains(['jsonl', 'csv'], options.format)) {
+            throw new errors.compileError('RT-INVALID-OPTION-COMBINATION',{
+                option: 'append',
+                rule: 'format="csv" or format="jsonl"'
+            });
+        }
+    }
+
+    _readHeaders() {
+        if (this.append && this.format === 'csv') {
+            return fs.accessAsync(this.filename)
+            .then(() => {
+                return new Promise((resolve, reject) => {
+                    // since the file exists lets make sure to let the csv
+                    // serializer to not send the headers
+                    this.serializerOptions.sendHeaders = false;
```

Comment:
This all really seems like it better belongs inside the serializer instead of here... obviously this is a little tricky since the current serializer API takes a stream and not the filename, but this really feels like a lot of logic that should be abstracted away in some better manner.